python/paddle/distributed/auto_parallel/static/engine.py

```diff
@@ -638,11 +638,10 @@ def _parallel_pir(self, mode):
         dist_program = paddle.base.libpaddle.pir.apply_mix2dist_pass(
             mix_fw_program
         )
-
-        # TODO(winter-wang) Step 1.2: pir backward
-        # with program_guard(dist_program):
-        #     params_grads = append_backward_pir(self._loss, parameter_list=self._parameter_list)
-
+        # Step 1.2: pir backward
+        if mode != "predict" and self._loss:
+            loss = dist_program.get_output_value_by_name(self._loss_names[0])
```

Comment:
what ablout multiple losses?

Reply:
The current PIR backward api only supports single losses situation.  Therefore, only the first loss is processed here. 
I will add annotations here in the next pr.